platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/dom/resolution/DefaultDocumentResolutionContainer.kt

```diff
@@ -309,7 +309,7 @@ class DocumentResolver(
             is ErrorReason.ValReassignment,
             ErrorReason.AccessOnCurrentReceiverOnlyViolation,
             is ErrorReason.NonReadableProperty,
-            is ErrorReason.AmbiguousImport -> error("not expected here")
```

Comment:
💭 This and the one for elements still should not be user-visible errors. The `not expected here` part was meant for us: if we ever see it, we should look into why it got reported in the first place, as it is a violation of the DOM contract.

Reply:
In this case, it seems that we have not updated the error mapping logic after we had allowed using named references (like `id`) on the RHS for the sake of supporting enums. 

Before that change, the whole assignment would have been transformed into an `ErrorNode` in the DOM because of unsupported syntax. 

Now that the syntax is supported, we should properly report this in the resolution results with a `PropertyNotAssignedReason` saying that it got resolved to another property (which is not allowed).

So what the test added in this PR asserts is not the desired behavior for this input. Should take care of properly mapping the error and changing the test?